apps/studio/components/interfaces/DiskManagement/DiskManagementReviewAndSubmitDialog.tsx

```diff
@@ -111,165 +121,211 @@ interface DiskSizeMeterProps {
+    planId: planId,
+    oldSize: formState.defaultValues?.totalSize || 0,
+    oldStorageType: formState.defaultValues?.storageType as DiskType,
+    newSize: getValues('totalSize'),
+    newStorageType: getValues('storageType') as DiskType,
     numReplicas,
   })
-
   const iopsPrice = calculateIOPSPrice({
     oldStorageType: form.formState.defaultValues?.storageType as DiskType,
     oldProvisionedIOPS: form.formState.defaultValues?.provisionedIOPS || 0,
     newStorageType: form.getValues('storageType') as DiskType,
     newProvisionedIOPS: form.getValues('provisionedIOPS'),
     numReplicas,
   })
-
   const throughputPrice = calculateThroughputPrice({
     storageType: form.getValues('storageType') as DiskType,
     newThroughput: form.getValues('throughput') || 0,
     oldThroughput: form.formState.defaultValues?.throughput || 0,
     numReplicas,
   })
 
+  const hasDiskConfigChanges =
+    form.formState.defaultValues?.provisionedIOPS !== form.getValues('provisionedIOPS') ||
+    (form.formState.defaultValues?.throughput !== form.getValues('throughput') &&
+      form.getValues('storageType') === 'gp3') ||
+    form.formState.defaultValues?.totalSize !== form.getValues('totalSize')
+
   return (
     <Dialog open={isDialogOpen} onOpenChange={setIsDialogOpen}>
       <DialogTrigger asChild>
         <ButtonTooltip
+          size="medium"
           htmlType="submit"
           type="primary"
           onClick={async (e) => {
             e.preventDefault()
             const isValid = await form.trigger()
             if (isValid) setIsDialogOpen(true)
           }}
-          disabled={isWithinCooldown || !isDirty}
+          disabled={disabled || !isDirty}
           tooltip={{
             content: {
               side: 'bottom',
               text: !canUpdateDiskConfiguration
                 ? 'You need additional permissions to update disk configuration'
-                : isWithinCooldown
-                  ? 'Currently within cooldown period from previous disk change'
+                : disabled
+                  ? 'Current form values are invalid'
                   : undefined,
             },
           }}
         >
           Review changes
         </ButtonTooltip>
       </DialogTrigger>
-      <DialogContent size="large">
+      <DialogContent className="min-w-[620px]">
         <DialogHeader>
           <DialogTitle>Review changes</DialogTitle>
-          <DialogDescription>
-            Disk configuration changes will be applied shortly once confirmed.
-          </DialogDescription>
+          <DialogDescription>Changes will be applied shortly once confirmed.</DialogDescription>
         </DialogHeader>
         <DialogSectionSeparator />
-
+        <div className="flex flex-col gap-2 p-5">
+          <DiskMangementRestartRequiredSection
+            visible={form.formState.defaultValues?.computeSize !== form.getValues('computeSize')}
+            title="Compute resizes trigger project restarts."
```

Comment:
this sentence makes sense, but I had to read it 2-3x to get it 
"Resizing your disk will trigger a project restart"  requires a bit less thinking? 
![screenshot-2024-10-30-at-15 24 16](https://github.com/user-attachments/assets/5618ac34-da46-4d12-af9d-c5d951dfa143)

Reply:
No, resizing your disk does not require a restart.

_Resizing your Compute_ does

Reply:
I agree with the grammar though

"Resizing your Compute will trigger a project restart" :D